libcxx/include/support/atomic/atomic_cuda.h

```diff
@@ -396,7 +396,7 @@ __host__ __device__
     _Tp __expected = __cxx_atomic_load(__a, memory_order_relaxed);
     _Tp __desired = __expected < __val ? __expected : __val;
 
-    while(__desired != __val &&
+    while(__desired == __val &&
```

Comment:
Either I've introduced a bug or there is something else wrong here. This was different from the implementation for max.